source/server/worker_impl.cc

```diff
@@ -18,8 +18,8 @@ WorkerPtr ProdWorkerFactory::createWorker(uint32_t index, OverloadManager& overl
                                           const std::string& worker_name) {
   Event::DispatcherPtr dispatcher(
       api_.allocateDispatcher(worker_name, overload_manager.scaledTimerFactory()));
-  return std::make_unique<WorkerImpl>(tls_, hooks_, std::move(dispatcher),
-                                      std::make_unique<ConnectionHandlerImpl>(*dispatcher, index),
+  auto conn_handler = std::make_unique<ConnectionHandlerImpl>(*dispatcher, index);
+  return std::make_unique<WorkerImpl>(tls_, hooks_, std::move(dispatcher), std::move(conn_handler),
```

Comment:
Fix a UD due to the order that `std::move(dispatcher)` and `std::make_unique<ConnectionHandlerImpl>(*dispatcher, index)` order is not deterministic (yet)

Reply:
@lambdai tiny nit: could you move this to another PR? This PR is great, thank you! Would be good to avoid conflating with any bug fix though.

Reply:
done